integration-tests/tests/src/tests/realm-constructor.ts

```diff
@@ -277,35 +342,74 @@ describe("Realm#constructor", () => {
     });
   });
 
-  // TODO: Next up is testRealmConstructorInMemory from realm-tests.js
-});
+  describe("in memory construction", () => {
+    it("data is shared among instances", () => {
+      // open in-memory realm instance
+      const realm1 = new Realm({ inMemory: true, schema: [TestObject] });
+      realm1.write(() => {
+        realm1.create("TestObject", [1]);
+      });
+      //@ts-expect-error TYPEBUG: isInMemory property does not exist
+      expect(realm1.isInMemory).to.be.true;
```

Comment:
Very strange that it's been overlooked. You could also just add it here: https://github.com/realm/realm-js/blob/test-migration-master/types/index.d.ts#L1133